src/toil/jobStores/aws/jobStore.py

```diff
@@ -1153,39 +1185,122 @@ def __getBucketVersioning(self, bucket):
+        if registry is None:
+            log.warn('Toil registry domain does not exist. Nothing will be updated after clean.')
 
+        # Now all variables pointing to job store components are None if and only if they do not
+        # exist.
+        i = 0
+        while True:
+            if filesBucket == filesDomain == jobsDomain is None:
+                log.error('No component of the job store exists. Nothing to clean.')
+            try:
+                if filesBucket is not None:
+                    log.debug('Attempting to deleting files bucket...')
+                    log.debug('Cancelling multipart uploads if any exist...')
+                    for attempt in retry_s3():
+                        with attempt:
+                            for upload in filesBucket.list_multipart_uploads():
+                                upload.cancel_upload()
+                    log.debug('No uploads remain.')
+                    log.debug('Attempting to delete keys in the files bucket if any exist...')
+                    if cls.__getBucketVersioning(filesBucket) in (True, None):
+                        for attempt in retry_s3():
+                            with attempt:
+                                for key in list(filesBucket.list_versions()):
+                                    filesBucket.delete_key(key.name, version_id=key.version_id)
+                    else:
+                        for attempt in retry_s3():
+                            with attempt:
+                                for key in list(filesBucket.list()):
+                                    key.delete()
+                    log.debug('No keys remain in the files bucket.')
+                    for attempt in retry_s3():
+                        with attempt:
+                            filesBucket.delete()
+                            filesBucket = None
+                    log.debug("Successfully deleted files bucket.")
+
+                for domain in (filesDomain, jobsDomain):
+                    if domain is not None:
+                        log.debug("Attempting to delete domain '%s'.", domain.name)
+                        for attempt in retry_sdb():
+                            with attempt:
+                                domain.delete()
+                                if domain is filesDomain:
+                                    filesDomain = None
+                                else:
+                                    jobsDomain = None
+                        log.debug('Successfully deleted domain.')
+                if registry is not None:
+                    registry.put_attributes(info.namePrefix, dict(exists=str(False)))
+                else:
+                    log.debug('No registry was updated since no registry domain was given.')
+            except Exception as e:
+                log.error('The following exception was raised during deletion:')
+                log.error(e, exc_info=True)
```

Comment:
Combine these two lines. The first argument to error() is a string not an exception. Look at logging.exception()